sender/firehose_sender.go

```diff
@@ -105,6 +108,48 @@ func (f *FirehoseSender) addKVMetaFields(fields map[string]interface{}) map[stri
 	return fields
 }
 
+func (f *FirehoseSender) calcDropLogProbability(fields map[string]interface{}) float64 {
+	logTime := fields["timestamp"].(time.Time)
+	delay := time.Since(logTime).Seconds() - 120
+	if delay <= 0 { // Don't drop logs with less than 2 minute delay
+		return 0
+	}
+
+	level := ""
+	if _, ok := fields["level"]; ok {
+		level = fields["level"].(string)
+	}
```

Comment:
These type assertions will panic if "timestamp" and "level" aren't the expected types. That's probably guaranteed by prior code but just checking?

Reply:
(example panic if level is an int https://play.golang.org/p/2Q7iHZOaCzG)

Reply:
I added `timestamp` to list of reserved fields, so it should always be of type `time.Time`: https://github.com/Clever/amazon-kinesis-client-go/blob/master/decode/decode.go#L18

I can add a check to for `level`.

Reply:
Updated in c6207831c6d270f40f707702db59f1ae7fcc1d39